crates/control/src/game_controller_state_filter.rs

```diff
@@ -156,14 +156,20 @@ fn next_filtered_state(
     config: &GameStateFilterParameters,
     ball_detected_far_from_any_goal: bool,
 ) -> State {
-    match (current_state, game_controller_state.game_state) {
-        (State::Finished, _) => State::Finished,
```

Comment:
I'm not a big fan of extending the match statement just for one additional distinction. What do you think about something like this instead?

```diff 
match (current_state, game_conntroller_state.game_state) {
-    (State::Finished, _) => State::Finished,
+    (State::Finished, _) => if matches!(game_controller_state.game_phase, GamePhase::PenaltyShootout { .. }) {
+        State::Set
+    } else {
+        State::Finished
+    },
```

Reply:
Yep sounds good, done